### PR TITLE
feature: Add tests for UI.

### PR DESCRIPTION
**Pull request by AKushch1337**
- Add tests for UI using JUnit4, Navigation testing library and Espresso
- Fix some small hardcoded strings

### CODE REVIEW

File: app/src/androidTest/java/com/example/bookish/UITest.kt
Comment (neliudochka) [inline app/src/androidTest/java/com/example/bookish/UITest.kt:1]: You can add more test scenarios. 
For example: 
- Test that settings in the Settings screen actually can be pressed. 
- Add a scenario when you hit the Back button when the back stack is empty. 
- Check that the list of books in the Library is actually scrollable.
- Test that the button in the Setting tab is clickable.

File: app/src/androidTest/java/com/example/bookish/UITest.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+@RunWith(JUnit4::class)
+class UITest {
+
+    @get:Rule
+    val composeTestRule = createComposeRule()
+
+    val LibraryButton = hasText("Library") and hasClickAction()
+    val ListsButton = hasText("Lists") and hasClickAction()
+    val SettingsButton = hasText("Settings") and hasClickAction()
```
Comment (neliudochka) [inline app/src/androidTest/java/com/example/bookish/UITest.kt]: Oh no, hardcoded strings...

File: app/src/androidTest/java/com/example/bookish/UITest.kt [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        composeTestRule.onNode(ListsButton).performClick()
+        composeTestRule.onNode(SettingsButton).performClick()
+        Espresso.pressBack()
+        composeTestRule.onNodeWithTag("lists_tag").assertIsDisplayed()
+    }
+
+    @Test
+    fun verify_if_back_stack_single_top_works_correctly() {
+        composeTestRule.onNode(ListsButton).performClick()
+        composeTestRule.onNodeWithTag("lists_tag").assertIsDisplayed()
```
Comment (neliudochka) [inline app/src/androidTest/java/com/example/bookish/UITest.kt]: Again, hardcode.
Put these tags to the constants and reuse them

File: app/src/androidTest/java/com/example/bookish/UITest.kt [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        composeTestRule.onNode(ListsButton).performClick()
+        composeTestRule.onNodeWithTag("lists_tag").assertIsDisplayed()
```
Comment (neliudochka) [inline app/src/androidTest/java/com/example/bookish/UITest.kt]: Okay, I won't comment on all of the hardcode, so just check the whole file.

File: app/src/main/java/com/example/bookish/composables/screens/Library.kt [hunk 4]
```diff
@@ -21,6 +23,7 @@ fun LibraryScreen() {
     Box(
         modifier = Modifier
             .fillMaxSize()
+            .semantics { testTag = "library_tag" }
```
Comment (neliudochka) [inline app/src/main/java/com/example/bookish/composables/screens/Library.kt]: hardcode

File: app/src/main/java/com/example/bookish/composables/screens/Lists.kt [hunk 5]
```diff
@@ -25,7 +27,8 @@ fun ListsScreen() {
             text = stringResource(R.string.lists),
             fontSize = 30.sp,
             fontWeight = FontWeight.Bold,
-            color = Color.White
+            color = Color.White,
+            modifier = Modifier.semantics { testTag = "lists_tag"  }
```
Comment (neliudochka) [inline app/src/main/java/com/example/bookish/composables/screens/Lists.kt]: hardcode? hardcode.

File: app/src/main/java/com/example/bookish/composables/screens/Settings.kt [hunk 6]
```diff
@@ -40,7 +42,7 @@ fun SettingsScreen() {
             Text(
                 text = stringResource(R.string.read_preferences),
                 style = MaterialTheme.typography.h5,
-                modifier = Modifier.padding(bottom = 8.dp)
+                modifier = Modifier.padding(bottom = 8.dp).semantics { testTag = "settings_tag" }
```
Comment (neliudochka) [inline app/src/main/java/com/example/bookish/composables/screens/Settings.kt]: oh wow, hardcode

**Review by neliudochka — CHANGES_REQUESTED** [review]
For the whole review you can, as always, check comments.
Here I want to point out again, that hardcode is bad. Do not hardcode your string values, it is painful to look at...
But this time no unnecessary imports. Good job!

File: app/src/androidTest/java/com/example/bookish/UITest.kt (hunk 1 above)
Comment (AKushch1337) [reply app/src/androidTest/java/com/example/bookish/UITest.kt]: Hardcode 😍

File: app/src/androidTest/java/com/example/bookish/UITest.kt (hunk 2 above)
Comment (AKushch1337) [reply app/src/androidTest/java/com/example/bookish/UITest.kt]: Hardcode 😍

File: app/src/androidTest/java/com/example/bookish/UITest.kt (hunk 3 above)
Comment (AKushch1337) [reply app/src/androidTest/java/com/example/bookish/UITest.kt]: Hardcode 😍

File: app/src/main/java/com/example/bookish/composables/screens/Library.kt (hunk 4 above)
Comment (AKushch1337) [reply app/src/main/java/com/example/bookish/composables/screens/Library.kt]: Hardcode 😍

File: app/src/main/java/com/example/bookish/composables/screens/Lists.kt (hunk 5 above)
Comment (AKushch1337) [reply app/src/main/java/com/example/bookish/composables/screens/Lists.kt]: Hardcode 😍

File: app/src/main/java/com/example/bookish/composables/screens/Settings.kt (hunk 6 above)
Comment (AKushch1337) [reply app/src/main/java/com/example/bookish/composables/screens/Settings.kt]: Hardcode 😍

File: app/src/androidTest/java/com/example/bookish/UITest.kt
Comment (AKushch1337) [reply app/src/androidTest/java/com/example/bookish/UITest.kt:1]: Can agree with all of them, except for the back stack empty one. From what I've learned UI test kit doesn't have access to the state of the app, and it's very hard to check if the app exited after we hit the back button, so I will omit this test case.

**Review by neliudochka — APPROVED** [review]
Okay, you can merge now.
P.S. Say no to hardcode!

**Comment by AKushch1337** [comment]
### I DID IT!
haardcodeshit

**Pull request merged**